pkg/resources/helpers_test.go

```diff
@@ -548,3 +548,23 @@ func queriedPrivilegesContainAtLeast(query func(client *sdk.Client, ctx context.
 		return nil
 	}
 }
+
+func updateAccountParameter(t *testing.T, client *sdk.Client, parameter sdk.AccountParameter, temporarily bool, newValue string) func() {
```

Comment:
I don't like that this function puts so many cleanup's on the stack. There is much bigger change that one of them will fail. We base it also on the LIFO behavior of the t.Cleanup (which may never change but can).

As I wrote to you earlier, I think that the `updateAccountParameterTemporarily` is sufficient here. The difference is, that you should invoke the `t.Cleanup` just in the first step modifying the param and ignore it in the subsequent steps, i.e.:
```
PreConfig: func() {
		revertParameter := updateAccountParameterTemporarily(t, client, sdk.AccountParameterDataRetentionTimeInDays, "1")
		t.Cleanup(revertParameter)
}
```

```
PreConfig: func() {
		_ := updateAccountParameterTemporarily(t, client, sdk.AccountParameterDataRetentionTimeInDays, "1")
		}
```

I think we should limit the number of additional helper functions and not add the ones similar to the already existing ones.

Reply:
I added a `temporarily` parameter earlier but for some reason didn't use it. I think I'll change it to return cleanup which feels more consistent with what we have in integration tests.

Reply:
Okay, I meant to reuse the same method or at least copy the existing one (because it is currently not importable), so that we do not have too many methods doing the exact same thing. Please just leave a TODO comment with SNOW-936093 and add it to the description of the issue (to merge this two eventually).